specification/communication/data-plane/CallingServer/preview/2022-04-07-preview/communicationservicescallingserver.json

```diff
@@ -1056,6 +1153,179 @@
+    "RecognizeInputType": {
+      "description": "Determines the type of the recognition.",
+      "enum": [
+        "dTMF"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecognizeInputType",
+        "modelAsString": true
+      }
+    },
+    "RecognizeRequest": {
+      "required": [
+        "recognizeOptions"
+      ],
+      "type": "object",
+      "properties": {
+        "recognizeInputType": {
+          "$ref": "#/definitions/RecognizeInputType"
+        },
+        "playSourceInfo": {
+          "$ref": "#/definitions/PlaySource",
+          "description": "The source of the audio to be played for recognition."
+        },
+        "stopCurrentOperations": {
+          "description": "If set recognize can barge into other existing queued-up/currently-processing requests.",
+          "type": "boolean"
+        },
+        "recognizeOptions": {
+          "$ref": "#/definitions/RecognizeOptions",
+          "description": "Defines options for recognition."
+        },
+        "operationContext": {
+          "description": "The value to identify context of the operation.",
+          "type": "string"
+        }
+      }
+    },
+    "RecognizeOptions": {
+      "type": "object",
+      "properties": {
+        "pausePlayOnResponse": {
+          "description": "Determines if we pause the prompt and start recognizing.",
+          "type": "boolean"
+        },
+        "initialSilenceTimeoutInSeconds": {
+          "format": "int32",
+          "description": "Time to wait for first input after prompt (if any) plays.",
+          "type": "integer"
+        },
+        "dtmfOptions": {
+          "$ref": "#/definitions/DtmfOptions",
+          "description": "Defines options for DTMF."
+        }
+      }
+    },
+    "DtmfOptions": {
+      "description": "Options for DTMF tone recognition",
+      "type": "object",
+      "properties": {
+        "collectTones": {
+          "$ref": "#/definitions/CollectTones",
+          "description": "Collects the number of digit in DTMF."
+        },
+        "dtfmToneProducer": {
+          "$ref": "#/definitions/CommunicationIdentifier",
+          "description": "Dtmf tone producer participant"
+        }
+      }
+    },
+    "CollectTones": {
+      "description": "Container for collecting tones for DTFM.",
+      "type": "object",
+      "properties": {
+        "interToneTimeoutInSeconds": {
+          "format": "int32",
+          "description": "Time to wait between DTMF inputs to stop recognizing.",
+          "type": "integer"
+        },
+        "maxTonesToCollect": {
+          "format": "int32",
+          "description": "Maximum number of DTMFs to be collected.",
+          "type": "integer"
+        },
+        "stopTones": {
+          "description": "List of tones that will stop recognizing.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CommunicationIdentifier": {
```

Comment:
Instead if inlining, we should reference the definitions `common.json` via relative path. An OpenApi Filter in the server code can be used to replace the model definition reference in the swagger.